pkg/autoscaler/autoscaler.go

```diff
@@ -345,6 +347,31 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 					expectedScaledMachineSet = machineSet
 				}
 			}
+
+			By("should have expected upstream annotations", func() {
+				Eventually(func() (bool, error) {
+					//checking for the keys of the newly added upstream annotations from the CAO
+					for _, machineSet := range machineSets {
+						if machineSet.Annotations == nil {
+							return false, nil
+						}
+						if _, exists := machineSet.Annotations[annotationsutil.CpuKey]; !exists {
+							return false, nil
+						}
+						if _, exists := machineSet.Annotations[annotationsutil.MemoryKey]; !exists {
+							return false, nil
+						}
+						if _, exists := machineSet.Annotations[annotationsutil.GpuCountKey]; !exists {
+							return false, nil
+						}
+						if _, exists := machineSet.Annotations[annotationsutil.GpuTypeKey]; !exists {
+							return false, nil
+						}
```

Comment:
i'm curious if we changed these to use the ginkgo matchers will we see better error output?

eg instead of `if` clauses, what if we did this:
```
Expect(machineSet.Annotations).To(HaveExistingField(annotationsutil.CpuKey))
```